packages/hash/api/src/model/knowledge/orgMembership.model.ts

```diff
@@ -0,0 +1,172 @@
+import { GraphApi } from "../../graph";
+import {
+  OrgMembershipModel,
+  EntityModel,
+  EntityModelCreateParams,
+  OrgModel,
+  UserModel,
+} from "..";
+import { extractBaseUri, workspaceAccountId } from "../util";
+import { WORKSPACE_TYPES } from "../../graph/workspace-types";
+
+export type OrgMembershipModelCreateParams = Omit<
+  EntityModelCreateParams,
+  "properties" | "entityTypeModel" | "accountId"
+> & {
+  responsibility: string;
+  org: OrgModel;
+};
+
+/**
+ * @class {@link OrgMembershipModel}
+ */
+export default class extends EntityModel {
+  /**
+   * Create a workspace OrgMembership entity.
+   *
+   * @param params.responsibility - the role of the user at the organization
+   */
+  static async createOrgMembership(
+    graphApi: GraphApi,
+    params: OrgMembershipModelCreateParams,
+  ) {
+    const { responsibility, org } = params;
+
+    const properties: object = {
+      [WORKSPACE_TYPES.propertyType.responsibility.baseUri]: responsibility,
+    };
+
+    const entityTypeModel = WORKSPACE_TYPES.entityType.orgMembership;
+
+    const entity = await EntityModel.create(graphApi, {
+      accountId: workspaceAccountId,
+      properties,
+      entityTypeModel,
+    });
+
+    await entity.createOutgoingLink(graphApi, {
+      linkTypeModel: WORKSPACE_TYPES.linkType.org,
+      targetEntityModel: org,
+    });
+
+    return new OrgMembershipModel({
+      accountId: workspaceAccountId,
+      entityId: entity.entityId,
+      version: entity.version,
+      entityTypeModel,
+      properties,
+    });
+  }
+
+  /**
+   * Get a workspace organization entity by its entity id.
+   *
+   * @param params.entityId - the entity id of the organization
+   */
+  static async getOrgMembershipById(
+    graphApi: GraphApi,
+    params: { entityId: string },
+  ): Promise<OrgMembershipModel | null> {
+    const entity = await EntityModel.getLatest(graphApi, {
+      // assumption: `accountId` of organizations is always the workspace account id
+      accountId: workspaceAccountId,
+      entityId: params.entityId,
+    });
+
+    if (
+      entity.entityTypeModel.schema.$id !==
+      WORKSPACE_TYPES.entityType.orgMembership.schema.$id
+    ) {
+      throw new Error(
+        `Entity with id ${params.entityId} is not a workspace org membership`,
+      );
+    }
+
+    return entity ? new OrgMembershipModel(entity) : null;
+  }
+
+  /**
+   * Get the org linked to the org membership.
+   */
+  async getOrg(graphApi: GraphApi): Promise<OrgModel> {
+    const { data: outgoingOrgLinks } = await graphApi.getLinksByQuery({
+      all: [
+        {
+          eq: [{ path: ["source", "id"] }, { literal: this.entityId }],
+        },
+        {
+          eq: [
+            { path: ["type", "uri"] },
+            {
+              literal: extractBaseUri(WORKSPACE_TYPES.linkType.org.schema.$id),
+            },
+          ],
```

Comment:
This can be replaced with `"versionedUri"`, once #1069 is merged 🚀

Reply:
Thanks for the reminder, addressed in https://github.com/hashintel/hash/pull/1068/commits/84c02cb8d3ec57b3a6b59e10286b4128a19eafe8